common/hyperscaler/gardener_client.go

```diff
@@ -0,0 +1,112 @@
+package hyperscaler
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kyma-project/kyma-environment-broker/common/gardener"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+)
+
+func (sp *GardenerClient) IsUsedByShoot(secretBinding *gardener.SecretBinding) (bool, error) {
+	shootlist, err := sp.gardenerClient.Resource(gardener.ShootResource).Namespace(sp.namespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return false, fmt.Errorf("listing Gardener shoots: %w", err)
+	}
+
+	for _, shoot := range shootlist.Items {
+		sh := gardener.Shoot{Unstructured: shoot}
+		if sh.GetSpecSecretBindingName() == secretBinding.GetName() {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func NewGardenerClient(gardenerClient dynamic.Interface, gardenerNamespace string) BindingsClient {
+	return &GardenerClient{
+		gardenerClient: gardenerClient,
+		namespace:      gardenerNamespace,
+	}
+}
+
+type GardenerClient struct {
+	gardenerClient dynamic.Interface
+	namespace      string
+}
+
+func (p *GardenerClient) GetSecretBinding(labelSelector string) (*gardener.SecretBinding, error) {
+	secretBindings, err := p.gardenerClient.Resource(gardener.SecretBindingResource).Namespace(p.namespace).List(context.Background(), metav1.ListOptions{
+		LabelSelector: labelSelector,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("listing secret bindings for LabelSelector: %s: %w", labelSelector, err)
+	}
+
+	if secretBindings != nil && len(secretBindings.Items) > 0 {
+		return &gardener.SecretBinding{Unstructured: secretBindings.Items[0]}, nil
+	}
+	return nil, nil
+}
+
+func (sp *GardenerClient) GetSecretBindings(labelSelector string) ([]unstructured.Unstructured, error) {
```

Comment:
This is kind of code duplication.
These two function differ slightly.
Both List matching secret bindings. Former gets 0th element. Latter returns entire list.
The result type could be adjusted.